gtsam/nonlinear/NonlinearOptimizerParams.h

```diff
@@ -71,6 +68,15 @@ class GTSAM_EXPORT NonlinearOptimizerParams {
   static Verbosity verbosityTranslator(const std::string &s) ;
   static std::string verbosityTranslator(Verbosity value) ;
 
+  /** Type for an optional user-provided hook to be called after each 
+   * internal optimizer iteration */
+  using IterationHook = std::function<
+    void(size_t /*iteration*/, double/*errorBefore*/, double/*errorAfter*/)>;
+
+  /** Optional user-provided iteration hook to be called after each 
+   * optimization iteration (Default: empty) */
+  IterationHook iterationHook;
```

Comment:
Can you please elaborate the docs more? One thing is to mention the expected arguments and their types. Perhaps even mention that the iteration hook is a `std::function` object?

Reply:
Done in the latest commit too.